src/WebJobs.Extensions.DurableTask/HttpRequestMessageExtensions.cs

```diff
@@ -68,7 +68,7 @@ public static NameValueCollection GetQueryNameValuePairs(this HttpRequestMessage
             var separator = '&';
 
             int l = (s != null) ? s.Length : 0;
-            int i = 0;
+            int i = 1; // skip the leading '?' character
```

Comment:
nit: s.TrimStart('?') feels more readable to me; is there a reason we don't do this?

Reply:
Yeah, mainly for low level performance. TrimStart('?') will allocate a new string object, and I was hoping to avoid that.